pkg/services/govmomi/vcenter/clone.go

```diff
@@ -350,12 +350,16 @@ func getDiskSpec(ctx *context.VMContext, devices object.VirtualDeviceList) ([]ty
 }
 
 func getDiskConfigSpec(disk *types.VirtualDisk, diskCloneCapacityKB int64) (types.BaseVirtualDeviceConfigSpec, error) {
-	if disk.CapacityInKB > diskCloneCapacityKB {
+	switch {
+	case diskCloneCapacityKB == 0:
+		// Use to the template disk capacity.
+	case diskCloneCapacityKB > 0 && diskCloneCapacityKB >= disk.CapacityInKB:
+		disk.CapacityInKB = diskCloneCapacityKB
+	case diskCloneCapacityKB > 0 && diskCloneCapacityKB < disk.CapacityInKB:
 		return nil, errors.Errorf(
 			"can't resize template disk down, initial capacity is larger: %dKiB > %dKiB",
 			disk.CapacityInKB, diskCloneCapacityKB)
 	}
```

Comment:
```suggestion
	if diskCloneCapacityKB > 0 {
		if disk.CapacityInKB > diskCloneCapacityKB {
			return nil, errors.Errorf(
				"can't resize template disk down, initial capacity is larger: %dKiB > %dKiB",
				disk.CapacityInKB, diskCloneCapacityKB)
		}
		disk.CapacityInKB = diskCloneCapacityKB
	}
```
Does this logic make sense?

Reply:
I tried this approach first, then tried the switch statement.

I found the switch statement to more clearly communicate that there are 3 possible cases (no size specified, valid size specified, invalid size specified).

Please let me know if you feel strongly that I should use the if, rather than the switch.

Reply:
My major concern was with the empty switch case but I see your point in the switch case laying out the 3 possible cases much more clearly.